packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -0,0 +1,860 @@
+    return OnPreviewStarted(true, "");
+  } else {
+    return OnPreviewStarted(false, "Preview already exists");
+  }
+
+  // Check MF_CAPTURE_ENGINE_PREVIEW_STARTED event handling for response
+  // process.
+  if (!preview_handler_->StartPreview(capture_engine_.Get(),
+                                      base_preview_media_type_.Get(),
+                                      capture_engine_callback_handler_.Get())) {
+    // Destroy preview handler on error cases to make sure state is resetted.
+    preview_handler_ = nullptr;
+    return OnPreviewStarted(false, "Failed to start video preview");
+  }
+}
+
+// Stops preview. Called by destructor
+// Use PausePreview and ResumePreview methods to for
+// pausing and resuming the preview.
+// Check MF_CAPTURE_ENGINE_PREVIEW_STOPPED event handling for response
+// process.
+void CaptureControllerImpl::StopPreview() {
+  assert(capture_engine_);
+
+  if (!IsInitialized() && !preview_handler_) {
+    return;
+  }
+
+  // Requests to stop preview.
+  preview_handler_->StopPreview(capture_engine_.Get());
+}
+
+// Marks preview as paused.
+// When preview is paused, captured frames are not processed for preview
+// and flutter texture is not updated
+void CaptureControllerImpl::PausePreview() {
+  assert(capture_controller_listener_);
+
+  if (!preview_handler_ && !preview_handler_->IsInitialized()) {
+    return capture_controller_listener_->OnPausePreviewFailed(
+        "Preview not started");
+  }
+
+  if (preview_handler_->PausePreview()) {
+    capture_controller_listener_->OnPausePreviewSucceeded();
+  } else {
+    capture_controller_listener_->OnPausePreviewFailed(
+        "Failed to pause preview");
+  }
+}
+
+// Marks preview as not paused.
+// When preview is not paused, captured frames are processed for preview
+// and flutter texture is updated.
+void CaptureControllerImpl::ResumePreview() {
+  assert(capture_controller_listener_);
+
+  if (!preview_handler_ && !preview_handler_->IsInitialized()) {
+    return capture_controller_listener_->OnResumePreviewFailed(
+        "Preview not started");
+  }
+
+  if (preview_handler_->ResumePreview()) {
+    capture_controller_listener_->OnResumePreviewSucceeded();
+  } else {
+    capture_controller_listener_->OnResumePreviewFailed(
+        "Failed to pause preview");
+  }
+}
+
+// Handles capture engine events.
+// Called via IMFCaptureEngineOnEventCallback implementation.
+// Implements CaptureEngineObserver::OnEvent.
+void CaptureControllerImpl::OnEvent(IMFMediaEvent* event) {
+  if (!IsInitialized() &&
+      capture_engine_state_ != CaptureEngineState::kInitializing) {
+    return;
+  }
+
+  GUID extended_type_guid;
+  if (SUCCEEDED(event->GetExtendedType(&extended_type_guid))) {
+    std::string error;
+
+    HRESULT event_hr;
+    if (FAILED(event->GetStatus(&event_hr))) {
+      return;
+    }
+
+    if (FAILED(event_hr)) {
+      // Reads system error
+      error = std::system_category().message(event_hr);
```

Comment:
Replace this line with:
```cc
_com_error err(event_hr);
error = Utf8FromUtf16(err.ErrorMessage());
```

You'll need to add this header at the top:
```cc
#include <comdef.h>
```

Reply:
Oh, yes, didn't thought about different languages at all. These errors should always be printed in english for now.
I would prefer having set of common camera error codes and messages across all platforms, at the moment there is no 
such list of error codes. With these users need to write custom error handling for each platform they are implementing their app. At the moment errors from all platforms might produce different error codes for a same problem. This might mean breaking change for all platforms though.
Currelty these errors are sen to ui with capture_error code.

Reply:
Fixed now as suggested.